### DISCUSSION TITLE
How to specify silent slash notes?

### DISCUSSION THREAD

**Discussion by infojunkie** [w3c-cg/musicxml#446] [Q&A] [answered]
I am trying to recreate lead sheets aka chord charts, such as those created by iReal Pro, in MusicXML. 

[Here's the demo](https://blog.karimratib.me/demos/musicxml/) of my library which converts the thousands of free iReal Pro lead sheets into MusicXML.

![screenshot-blog karimratib me-2022 02 15-17_05_03](https://user-images.githubusercontent.com/92977/154176582-bc420477-86c5-40e0-a05b-e53c84eb2e60.png)

My question is about the semantics of the `measure-style > slash` element, which I use to specify the duration of each chord: does this element imply that the corresponding slashed notes in the measure should NOT be played? Otherwise, how can one specify that slashed notes should be kept silent by the audio player (or replaced with their chord)?

As an example of interpretation of `measure-style > slash` in the wild, MuseScore does mute any note that appears in a measure containing such an element.

Thanks,
Karim

**Comment by mdgood** [comment]
The [`<slash>`](https://www.w3.org/2021/06/musicxml40/musicxml-reference/elements/slash/) measure-style does not imply silence. Often people will prepare some sample playback and hide the exact notation with a slash style.

I think the simplest way to not have playback would be to just use measure rests ([`<rest measure="yes"/>`](https://www.w3.org/2021/06/musicxml40/musicxml-reference/elements/rest/)) for each measure in the slash style. Alternatively you could add a [`<cue/>`](https://www.w3.org/2021/06/musicxml40/musicxml-reference/elements/cue/) element to each note. The `<cue/>` element is what specifies that a note is silent. If you want the chord to play back, just enter a whole note version of the chord in the music. Let me know if I am misunderstanding something here.

**Reply by infojunkie** [reply]
> If you want the chord to play back, just enter a whole note version of the chord in the music.

The style of presentation of lead sheets is to show slash notes indicating the rhythm, or the tune's melody line, but not the chord tones - which are only shown as chord names. Did I misunderstand your comment?

`cue` might be of help, but I think that semantically the cue notes means something different.

Essentially, I'm trying to recreate ["Chord charts"](https://en.wikipedia.org/wiki/Chord_chart).

**Comment by mdgood** [comment, marked as answer]
Ah, that wasn't in your example. To get that appearance, make sure you use `<slash use-stems="yes"/>`. In that case, you would either need to specify the chords in the rhythm you want, or use a `<cue/>` element on each note within the style.

The `<cue/>` element is how you represent silent notes (vs. rests) in MusicXML. The element is named after the most common use case for these silent notes, but it is used in other situations as well.

**Reply by infojunkie** [reply]
Ok, thanks. I'll specify `cue` and I hope that the various players honor it.

**Reply by infojunkie** [reply]
Still, I'd like to point out that cue notes are meant to be displayed in smaller size. Slash notes on chord charts are not. 

Would an attribute on `sound` other than `mute` make sense here?

**Comment by mdgood** [comment]
You can indicate full-size cue notes with the [`<type>`](https://www.w3.org/2021/06/musicxml40/musicxml-reference/elements/type/) element by using the size attribute: `<type size=full>quarter</type>`.

**Comment by infojunkie** [comment]
Trying to implement this, I ran into a situation where I need to generate tied notes (that are still cues) - because the desired duration is not expressible otherwise. It seems `cue` and `tie` don't mix in the schema?

**Reply by mdgood** [reply]
Correct. The `<tie>` element is for playback, and `<cue>` notes don't play. While normal notes use both the `<tie>` and `<tied>` elements, `<cue>` notes use only the `<tied>` element.

**Reply by infojunkie** [reply]
Makes sense! Thanks.
